e2e-tests/spring-rest-mongo/pom.xml

```diff
@@ -94,6 +94,13 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <reuseForks>false</reuseForks>
```

Comment:
add comment to explain reason for this specific option

Reply:
I post the comment also here: 

There is an issue that arises when running the MongoEMGenerationTest after other Mongo tests. The problem is that MappingMongoEntityInformationReplacement does not work in that case, causing the test to fail. This replacement relies on the flag "instrumentMR_MONGO" to be true when the SUT starts. However, in the other tests, except for MongoEMGenerationTest, "instrumentMR_MONGO" is set to false when the SUT starts. It seems that when the replacement does not occur in the other tests, some component is left in an inconsistent state, or there may be a related cache that prevents the replacement from happening in the MongoEMGenerationTest. The issue can be resolved by using separate JVM processes for each test, which solves the problem.

Reply:
hi @hghianni ,
 isn't this the same issue we discussed once, and that I did refer to how to address with example from `ExternalServiceMockingFlakyEMTest`? ie:

```
 @BeforeAll
    public static void initClass() throws Exception {
        ServiceController serviceController = new ServiceController();
        EMConfig config = new EMConfig();
        config.setInstrumentMR_NET(true);
        SpringTestBase.initClass(serviceController, config);
    }
```

can you please try the equivalent solution for MongoDB? to see if indeed it is the same problem or not. recall that when running tests from IDE those settings in maven plugins would be not applied by default

Reply:
Hi @arcuri82 , MongoEMGenerationTest is already configured like that. If not, MappingMongoEntityInformation replacement would not happen, and the test would fail as instrumentMR_MONGO would be set as false when starting the controller. This is the only test that needs this replacement, so the others are not configured the same way. If I configure all like this, all tests will pass, but it shouldn't be required. Maybe is it a better solution that forcing all to run on different processes. It seems like if the replacement happened in the first test run, then it will happen in the MongoEMGeneration test. If it didn't happen, it will not happen in the MongoEMGeneration test. Is there a cache related to replacements that may prevent them if they previously were not possible?.  I was able to reproduce the issue both with Maven and the IDE by running all tests in the same process.

Reply:
hi @hghianni the only cache is the classloader. a class can be loaded only once in a classloader. if for any reason a class is loaded before the flag for the instrumentation is set, then it cannot be changed.